src/neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -144,6 +149,23 @@ public Witness[] Witnesses
             set { witnesses = value; _size = 0; }
         }
 
+        static Transaction()
+        {
+            using ScriptBuilder sb = new ScriptBuilder();
+            sb.EmitAppCall(NativeContract.Oracle.Hash, "finish");
+            oracleResponseScript = sb.ToArray();
+        }
+
+        private bool CheckOracleResponse(StoreView snapshot)
+        {
+            OracleResponse response = Attributes.OfType<OracleResponse>().FirstOrDefault();
+            if (response is null) return true;
+            OracleRequest request = NativeContract.Oracle.GetRequest(snapshot, response.Id);
+            if (request is null) return false;
+            Transaction request_tx = snapshot.GetTransaction(request.Txid);
```

Comment:
```suggestion
            Transaction request_tx = snapshot.GetTransaction(request.Txid);
            if (request_tx is null) return false;
```

Reply:
It can't be `null`. Since there is a request, the transaction must exist.

Reply:
If the oracle response it's evil, it can be done

Reply:
No, It can't be. If the oracle response is evil, the request could be `null`. But if the request is not `null`, then the transaction can't be `null`. Because the request is written by the transaction by calling `Request()`. This can't be modified.